### PR TITLE
comp-fork update

### PR DESCRIPTION
**Pull request by dmelotik**
Organize mapping.ts code, add some constants, update dependecies

### CODE REVIEW

**Review by 0xbe1 — COMMENTED** [review]
lgtm! as long as existing compound forks run without problem. Have you tried any of the existing subgraph?

File: subgraphs/compound-forks/src/mapping.ts [hunk 1]
```diff
@@ -1318,11 +1381,100 @@ function updateProtocol(comptrollerAddr: Address): void {
...
+    protocol.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeDepositUSD = BIGDECIMAL_ZERO;
+    protocol.totalBorrowBalanceUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeBorrowUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeLiquidateUSD = BIGDECIMAL_ZERO;
+
+    protocol._priceOracle = "";
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts:1418]: good idea, moved

File: subgraphs/compound-forks/src/mapping.ts [hunk 2]
```diff
@@ -866,7 +897,7 @@ function snapshotMarket(
  * @param blockTimestamp
  * @returns
  */
-function snapshotFinancials(
+export function snapshotFinancials(
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts:900]: for fuse I had to "override" some functions so I used similar logic where I could. So from my fuse/mapping.ts I am calling this function so I don't have to repeat logic. Maybe to be consistent all of the non-helper functions should be `export` functions in this file?

File: subgraphs/compound-forks/src/mapping.ts [hunk 3]
```diff
@@ -281,6 +238,29 @@ export function _handleNewPriceOracle(
...
+//
+// event.params
+//  - cToken: Address
+//  - action: string
+//  - pauseState: boolean
+export function _handleActionPaused(event: ActionPaused1): void {
+  let marketID = event.params.cToken.toHexString();
+  let market = Market.load(marketID);
+  if (!market) {
+    log.warning("[handleMint] Market not found: {}", [marketID]);
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts:251]: fixed

File: subgraphs/compound-forks/src/mapping.ts [hunk 4]
```diff
@@ -824,8 +859,6 @@ function snapshotMarket(
   dailySnapshot.rewardTokenEmissionsAmount = market.rewardTokenEmissionsAmount;
   dailySnapshot.rewardTokenEmissionsUSD = market.rewardTokenEmissionsUSD;
   dailySnapshot.rates = market.rates;
-  dailySnapshot.rewardTokenEmissionsAmount = market.rewardTokenEmissionsAmount;
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts:827]: it was in there twice

**Comment by dmelotik** [comment]
> lgtm! as long as existing compound forks run without problem. Have you tried any of the existing subgraph?

Tested with cream-finance on mainnet, no problems. lmk if you think I should test more!

File: subgraphs/compound-forks/src/mapping.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xbe1) [reply subgraphs/compound-forks/src/mapping.ts:900]: let's only add `export` whenever needed for now :) and we can add more later

File: subgraphs/compound-forks/src/mapping.ts [hunk 5]
```diff
@@ -281,6 +238,29 @@ export function _handleNewPriceOracle(
   protocol.save();
 }
 
+//
+//
+// event.params
+//  - cToken: Address
+//  - action: string
+//  - pauseState: boolean
+export function _handleActionPaused(event: ActionPaused1): void {
```
Comment (0xbe1) [inline subgraphs/compound-forks/src/mapping.ts:247]: I think we need to call _handleActionPaused in every subgraph

File: subgraphs/compound-forks/src/mapping.ts [hunk 6]
```diff
@@ -1138,6 +1191,12 @@ function updateMarket(
     underlyingToken.decimals
   );
 
+  log.warning("Updated price for {}, mantissa: {}, price: ${}", [
```
Comment (0xbe1) [inline subgraphs/compound-forks/src/mapping.ts:1194]: shall we remove?

File: subgraphs/compound-forks/src/mapping.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts:900]: sounds good- resolving

File: subgraphs/compound-forks/src/mapping.ts (hunk 6 above)
Comment (dmelotik) [reply subgraphs/compound-forks/src/mapping.ts]: good catch - I checked for more testing logs, and they should all be gone now

File: subgraphs/compound-forks/cream-finance/package.json [hunk 7]
```diff
@@ -1,4 +1,5 @@
 {
+  "private": true,
```
Comment (0xbe1) [inline subgraphs/compound-forks/cream-finance/package.json:2]: what is this for?

**Review by 0xbe1 — COMMENTED** [review]
l left 2 more small comments. otherwise, lgtm!

File: subgraphs/compound-forks/cream-finance/package.json (hunk 7 above)
Comment (dmelotik) [reply subgraphs/compound-forks/cream-finance/package.json:2]: without this I wasn't able to run the scripts -- on my end scripts either have to have that tag or have the license and all the other headers

Comment (0xbe1) [reply subgraphs/compound-forks/cream-finance/package.json:2]: what is the error log without "private"?

File: subgraphs/compound-forks/src/constants.ts [hunk 8]
```diff
@@ -93,6 +101,7 @@ export const mantissaFactor = 18;
 export const cTokenDecimals = 8;
 export const mantissaFactorBD = exponentToBigDecimal(mantissaFactor);
 export const cTokenDecimalsBD = exponentToBigDecimal(cTokenDecimals);
+export const initialExchangeRate = BigDecimal.fromString(".02");
```
[the remark this answers is not in the document]
Comment (dmelotik) [reply subgraphs/compound-forks/src/constants.ts]: removed and init to 0 for more universal mappings

File: subgraphs/compound-forks/cream-finance/package.json (hunk 7 above)
Comment (dmelotik) [reply subgraphs/compound-forks/cream-finance/package.json:2]: well i guess it does work (i just get a license warning). I think my file tree wasn't updating right away. 

Removed

**Pull request merged**